x/ccv/provider/keeper/permissionless.go

```diff
@@ -0,0 +1,440 @@
+package keeper
+
+import (
+	errorsmod "cosmossdk.io/errors"
+	storetypes "cosmossdk.io/store/types"
+	"encoding/binary"
+	"fmt"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/interchain-security/v5/x/ccv/provider/types"
+	"time"
+)
+
+// ConsumerPhase captures the phases of a consumer chain according to `docs/docs/adrs/adr-018-permissionless-ics.md`
+type ConsumerPhase byte
+
+const (
+	Registered ConsumerPhase = iota
+	Initialized
+	Launched
+	Stopped
+)
+
+var DefaultUpdateRecord = types.ConsumerUpdateRecord{
```

Comment:
Removed `DefaultUpdateRecord` altogether.

> Maybe it makes sense to have all these records in a different proto file than tx.proto, e.g., provider.proto and then have types/provider.go.

This is an option. Had all the records under `tx.proto` because they're part of the messages but we could have them under `provider.proto`. Let me know if you want them this way.

Reply:
@insumity The records are used for storing stuff in state. The messages are used for transactions. For simplicity, the messages contain the record as a field to avoid duplicating code. So IMO, records should go to provider.proto (which contains definition of data structures needed by the provider) and messages go in tx.proto.